kamon-core/src/main/scala/kamon/ModuleSupervisor.scala

```diff
@@ -0,0 +1,49 @@
+package kamon
+
+import _root_.akka.actor
+import _root_.akka.actor._
+import kamon.ModuleSupervisor.CreateModule
+
+import scala.concurrent.{ Future, Promise }
+import scala.util.Success
+
+object ModuleSupervisor extends ExtensionId[ModuleSupervisorExtension] with ExtensionIdProvider {
+
+  def lookup(): ExtensionId[_ <: actor.Extension] = ModuleSupervisor
+  def createExtension(system: ExtendedActorSystem): ModuleSupervisorExtension = new ModuleSupervisorExtensionImpl(system)
+
+  case class CreateModule(name: String, props: Props, childPromise: Promise[ActorRef])
+}
+
+trait ModuleSupervisorExtension extends actor.Extension {
+  def createModule(name: String, props: Props): Future[ActorRef]
+}
+
+class ModuleSupervisorExtensionImpl(system: ExtendedActorSystem) extends ModuleSupervisorExtension {
+  import system.dispatcher
+  private val supervisor = system.actorOf(Props[ModuleSupervisor], "kamon")
+
+  def createModule(name: String, props: Props): Future[ActorRef] = Future {} flatMap { _: Unit ⇒
```

Comment:
why :Future[ActorRef] = Future{} flatMap { _: Unit => instead of Future[ActorRef] ?